src/lib/stores/User.ts

```diff
@@ -134,14 +134,35 @@ export async function login(token: string) {
   }
 }
 
-export async function logout() {
+export async function logout(authProvider: AuthProvider | undefined) {
   if (browser) {
     const token = localStorage.getItem('token');
-    token && api.get('/psama/logout');
+    token && (await api.get('/psama/logout'));
     token && localStorage.removeItem('token');
   }
-  user.set({});
-  goto('/login');
+
+  // get the auth provider
+  if (authProvider) {
+    authProvider
+      .logout()
+      .then((redirect) => {
+        user.set({});
+        if (typeof redirect === 'string') {
+          location.replace(redirect);
+        } else {
+          // If no redirect is provided, go to the login page
+          console.error('Error logging out: ' + redirect);
+          goto('/login');
+        }
+      })
+      .catch((error) => {
+        console.error('Error logging out: ' + error);
+        goto('/login');
+      });
```

Comment:
I think we always want to go to the login page when the user gets logged out. What is the redirect for?

Reply:
No, in this case we do not. We must follow the redirect. It will send the user to the RAS session logout page, they will then be redirected to OKTA, and finally OKTA will redirect them to our login page.

Reply:
Ah, right, the triple redirect thing